### PR TITLE
registration form (solves: #6)

### PR DESCRIPTION
**Pull request by stankasvecova**
solves: #6

### CODE REVIEW

File: registration.html
Comment (BTBTravis) [inline registration.html]: Please cleanup `<<<<<` git conflict leftovers.

File: registration.html
Comment (BTBTravis) [inline registration.html]: why is there two forms?

File: registration.html
Comment (BTBTravis) [inline registration.html]: please move script into a js file and remember to wrap it in a function scope. 

```js
(function () {
    function autocomplete(inp, arr) {
    ...
})();
```

File: registration.html
Comment (BTBTravis) [inline registration.html]: I would advise against bootstrap. Its better to learn how to quickly style things yourself. Truly mastering grid and flex will take you much further, might as well start now. 

Fully understand 
https://css-tricks.com/snippets/css/complete-guide-grid/

and 

https://css-tricks.com/snippets/css/a-guide-to-flexbox/

File: registration.html
Comment (BTBTravis) [inline registration.html]: move styles to a css file

File: registration.html
Comment (BTBTravis) [inline registration.html]: don't inline styles. use css file.

File: registration.html
Comment (BTBTravis) [inline registration.html]: I think type number will suit you here

File: registration.html
Comment (BTBTravis) [inline registration.html]: this bloat html is another reason to avoid bootstrap

File: registration.html
Comment (BTBTravis) [inline registration.html]: type email will suit you here

File: registration.html
Comment (BTBTravis) [inline registration.html]: also type email here

File: registration.html
Comment (BTBTravis) [inline registration.html]: type password would be better

File: registration.html
Comment (BTBTravis) [inline registration.html]: there are bad variable names. It should be clear what something is by it's name. what is a or b? Only exception is `i` inside a for loop

File: registration.html
Comment (BTBTravis) [inline registration.html]: very strong feeling you copied and pasted this code without understanding it...

**Review by BTBTravis — CHANGES_REQUESTED** [review]
Nice start on the registration page @stankasvecova ! Left some comments of things that can be improved before we move on.

File: registration.html
Comment (stankasvecova) [reply registration.html]: Yes, I used the grid at the online store, and now I wanted to give Bootstrap a go. Didn't know it was not advisable. I will use grid & flex from now onwards but does this bootstrap here need to be removed too?

File: registration.html
Comment (stankasvecova) [reply registration.html]: Just this time, as it was just one compact thing ' registration', I thought I would have it all in one file.
Happy to split it into separate .css and .js.

File: registration.html
Comment (stankasvecova) [reply registration.html]: OK

File: registration.html
Comment (stankasvecova) [reply registration.html]: will amend.

File: registration.html
Comment (stankasvecova) [reply registration.html]: OK

File: registration.html
Comment (stankasvecova) [reply registration.html]: will amend.

File: registration.html
Comment (stankasvecova) [reply registration.html]: will amend.

File: registration.html
Comment (stankasvecova) [reply registration.html]: will amend.

File: registration.html
Comment (stankasvecova) [reply registration.html]: will do.

File: registration.html
Comment (stankasvecova) [reply registration.html]: Well, I tried to understand that code and went thru it a few times. 
There is the concept of .this (and other stuff) that seemed too difficult and so I tried whether it would work anyway, and it did. I planned to go back to it, and analyse it further to understand it. Happy to improve it, once I do.

File: registration.html
Comment (stankasvecova) [reply registration.html]: See above.

**Review by BTBTravis — CHANGES_REQUESTED** [review]
The input types were changed in the wrong manner

```html
<input type="number" ...
<input type="email" ...
```
more here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input

File: registration.html
Comment (BTBTravis) [reply registration.html]: ✅

File: registration.html
Comment (BTBTravis) [inline registration.html]: Don't seen need for jQuery on the site or bootstrap on the site.

File: registration.html
Comment (BTBTravis) [inline registration.html]: html indentation is messed up here

File: registration.html
Comment (BTBTravis) [inline registration.html]:
```html
<input type="email" ...
``` 
will give us some decent native validation of the form feild

File: registration.html
Comment (MrTim) [inline registration.html]: if you really want to have this Agree check => you need to provide those T&C

File: registration.html
Comment (MrTim) [inline registration.html]: just age

File: registration.html
Comment (MrTim) [inline registration.html]: where these strange id's are coming from?

**Review by MrTim — CHANGES_REQUESTED** [review]

File: registration.html
Comment (BTBTravis) [reply registration.html]: Was reading https://www.swyx.io/writing/js-third-age/ this morning. Made me think back to the jQuery days. I would focus on new tech as you continue to expand your JavaScript skills.

**Review by stankasvecova — COMMENTED** [review]
**My last push should cover all Ivan's comments:**
1. why strange IDs: as I believed they would be used in back-end for some validation, but true, don't need them this complicated. 
2. T&C will be added once I have it. For now an empty <a href=""></a>

I did not change bootstrap html into grid yet, i wasn't aware of it being not a good choice. I would continue with the rest of the pages with grid though. Let me know if it needs to be changed also in the registration.

File: registration.html
Comment (BTBTravis) [reply registration.html]: For me it is not bootstrap vs grid & flex box. It is a css framework vs custom styles. I think its better to learn styling everything yourself because if you truly want to build something custom no css framework will do and it is better to write the styles yourself. 

For this page I won't block however lets move on.

File: registration.html
Comment (BTBTravis) [inline registration.html]: this is not the place to attach scripts and I don't think this script exist but we can fix that in future prs

**Review by BTBTravis — DISMISSED** [review]
Looking good!

**Review by MrTim — DISMISSED** [review]
Looks good! Please merge this one.

**Review by stankasvecova — COMMENTED** [review]
Before I merge this one, I wanted to add the autocomplete javascript first. 
I understand the code (except a part of it which I'm attaching at the bottom of this message). The parts I did not understand, I experimented by leaving it out and it still works. I renamed variables into something more meaningful. I made the code shorter and simpler without compromising the functionality - it does work. Are you happy if it stays like this?
---
the parts of the code I didn't get:
inp.addEventListener("keydown", function(e) {  
  var x = document.getElementById(this.id + "autocomplete-list"); 
  if (x) x = x.getElementsByTagName("div");
  if (e.keyCode == 40) {
    currentFocus++;
    addActive(x);
  } else if (e.keyCode == 38) {
    currentFocus--;
    addActive(x);
  } else if (e.keyCode == 13) {
    e.preventDefault();
    if (currentFocus > -1) {
      if (x) x[currentFocus].click();
    }
  }
function addActive(x) {
if (!x) return false;
removeActive(x);
if (currentFocus >= x.length) currentFocus = 0;
if (currentFocus < 0) currentFocus = (x.length - 1);
x[currentFocus].classList.add("autocomplete-active");
}
function removeActive(x) {
for (var i = 0; i < x.length; i++) {
  x[i].classList.remove("autocomplete-active");
}
}

File: registration.css
Comment (MrTim) [inline registration.css]: looks like your not using this style. If it's true => remove it

File: registration.html
Comment (MrTim) [inline registration.html]: you can put css file under css folder which is already in your project

File: registration.html
Comment (MrTim) [inline registration.html]: the same with js files - please put all js files under js folder, so you would have more solid structure for the project

File: registration.html
Comment (MrTim) [inline registration.html]: where is btn-danger style is coming from? I can't find it...

File: registration.html
Comment (MrTim) [inline registration.html]: which action you want to perform on submitting the form?

File: registration.js
Comment (MrTim) [inline registration.js]: two things:

1. you have problem with indentation. Please move all function content right
2. why it's `var`?

File: registration.js
Comment (MrTim) [inline registration.js]: do you understand what `this` keyword is stands for in here?

**Review by MrTim — CHANGES_REQUESTED** [review]

**Comment by MrTim** [comment]
> ## Before I merge this one, I wanted to add the autocomplete javascript first.
> I understand the code (except a part of it which I'm attaching at the bottom of this message). The parts I did not understand, I experimented by leaving it out and it still works. I renamed variables into something more meaningful. I made the code shorter and simpler without compromising the functionality - it does work. Are you happy if it stays like this?
> 
> the parts of the code I didn't get:
> inp.addEventListener("keydown", function(e) {
> var x = document.getElementById(this.id + "autocomplete-list");
> if (x) x = x.getElementsByTagName("div");
> if (e.keyCode == 40) {
> currentFocus++;
> addActive(x);
> } else if (e.keyCode == 38) {
> currentFocus--;
> addActive(x);
> } else if (e.keyCode == 13) {
> e.preventDefault();
> if (currentFocus > -1) {
> if (x) x[currentFocus].click();
> }
> }
> function addActive(x) {
> if (!x) return false;
> removeActive(x);
> if (currentFocus >= x.length) currentFocus = 0;
> if (currentFocus < 0) currentFocus = (x.length - 1);
> x[currentFocus].classList.add("autocomplete-active");
> }
> function removeActive(x) {
> for (var i = 0; i < x.length; i++) {
> x[i].classList.remove("autocomplete-active");
> }
> }

regarding this:
first - you can better attach code samples in github comments. Please take a look at this link https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks

About question itself
`keyCode == 38` is an arrow key `up` and `keyCode == 40` is an arrow key `down` which gives a user possibility to choose needed country with up down keys without a mouse. 
If you want to use this code - it's fine. But please fix all variable keywords. It would be nice to not move the content of the main page, but I think for now we can leave it as it is.

File: registration.css
Comment (stankasvecova) [reply registration.css]: Removed

File: registration.html
Comment (stankasvecova) [reply registration.html]: Will do.

File: registration.html
Comment (stankasvecova) [reply registration.html]: will do.

File: registration.html
Comment (stankasvecova) [reply registration.html]: From bootstrap pre-set styles in the <link ref=""> on line 8. But it's true, i don't need to have it doubled, btn style is enough, so btn-danger style is now removed.

File: registration.html
Comment (stankasvecova) [reply registration.html]: Posting the inputed data into the process.php file.

File: registration.js
Comment (stankasvecova) [reply registration.js]: ran auto-format function on it so hopefully fixed. 
'var's - didn't poke my eye, my bad. Changed into 'let's.

File: registration.js
Comment (stankasvecova) [reply registration.js]: yes, we are setting the "id" for the countryDiv, and this.id relates to the current created countryDiv?

File: registration.html
Comment (stankasvecova) [reply registration.html]: Does exist now and script attached correctly now. Thanks

File: docs/hobbies.html
Comment (BTBTravis) [inline docs/hobbies.html]: First this is in the wrong folder. `/docs` is for documentation markdown files and images for those files. Also this pull request is only about registration form please remove this file from the pull request. You may submit it in another

File: js/registration.js
Comment (BTBTravis) [inline js/registration.js:1]: I think it is a mistake to include this javascript. #6 asks for a static html css registration form. Lets keep things simple and expand on them with javascript in future PRs

**Review by BTBTravis — CHANGES_REQUESTED** [review]
hey @stankasvecova lets reduce the size of this pr so we can get it merged and move on

File: js/registration.js
Comment (MrTim) [reply js/registration.js:1]: It's too late to remove the js from here. I did review already and need to do a final review and it would be merged.

File: docs/hobbies.html
Comment (MrTim) [reply docs/hobbies.html]: @stankasvecova please move it into html folder. 
And this part has nothing to do with registration, so please remove it from here and continue with it in different branch.

File: docs/registration.html
Comment (MrTim) [inline docs/registration.html]: Please move this one into `html` forlder.

**Review by MrTim — CHANGES_REQUESTED** [review]
Please remove hobbies from this pr and put registration.html in html folder and after that this PR will be ready to merge.

File: docs/hobbies.html
Comment (stankasvecova) [reply docs/hobbies.html]: Done

File: docs/registration.html
Comment (stankasvecova) [reply docs/registration.html]: done

**Review by MrTim — APPROVED** [review]
Nice, now it's better!
Only thing you need to fix is the css and logo imports
`<link rel="stylesheet" href="registration.css">` because now they are in different folders.
But you can do it and after merge!

**Review by BTBTravis — APPROVED** [review]
👍

**Pull request merged**
